src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -538,12 +541,19 @@ event::ptr primitive_inst::realloc_if_needed() {
 
                 _outputs[0] = variable.get_memory();
                 // To record shape predictor
-                auto prealloc_info = sp.predict_preallocation_shape(id(), _impl_params->output_layouts[0], true);
+                sp.predict_preallocation_shape(id(), _impl_params->output_layouts[0], true, 0);
+                if (_outputs.size() > 1) {
+                    sp.predict_preallocation_shape(id(), _impl_params->output_layouts[1], true, 1);
```

Comment:
Is not necessary to run predict_preallocaction_shape() for 3rd output layout when number of outputs are three?

Reply:
This part is for the kv cache op which has only 2 outputs at max.

Reply:
Okay, I see. thanks.